dtfs-central-api/src/services/facility-utilisation-data/facility-utilisation-data.service.test.ts

```diff
@@ -0,0 +1,203 @@
+
+        // Assert
+        expect(mockSave).toHaveBeenCalledTimes(1);
+        const expectedEntities = [
+          FacilityUtilisationDataEntity.create({
+            id: facilityIdWithoutExistingUtilisationData,
+            reportPeriod: previousReportPeriod,
+            requestSource,
+            utilisation,
+            fixedFee,
+          }),
+        ];
+        expect(mockSave).toHaveBeenCalledWith(FacilityUtilisationDataEntity, expectedEntities, { chunk: 100 });
+      });
+    });
+
+    describe('when all facility ids are not associated with existing FacilityUtilisationDataEntities', () => {
+      const firstFacilityId = '01';
+      const secondFacilityId = '02';
+      const facilityIds = new Set([firstFacilityId, secondFacilityId]);
+
+      beforeEach(() => {
+        mockFindBy.mockResolvedValue([]);
+      });
+
+      it('should create new FacilityUtilisationDataEntities for all facilities', async () => {
+        // Arrange
+        const firstUtilisation = 111.11;
+        const secondUtilisation = 222.22;
+        const firstFixedFee = 333.33;
+        const secondFixedFee = 444.44;
+
+        const requestSource = aDbRequestSource();
+        const reportPeriod = { start: { month: 1, year: 2023 }, end: { month: 2, year: 2023 } };
+        const previousReportPeriod = { start: { month: 11, year: 2022 }, end: { month: 12, year: 2022 } };
+
+        when(calculateInitialUtilisationAndFixedFee).calledWith(firstFacilityId).mockResolvedValue({ fixedFee: firstFixedFee, utilisation: firstUtilisation });
+        when(calculateInitialUtilisationAndFixedFee)
+          .calledWith(secondFacilityId)
+          .mockResolvedValue({ fixedFee: secondFixedFee, utilisation: secondUtilisation });
+
+        jest.mocked(getPreviousReportPeriod).mockResolvedValue(previousReportPeriod);
+
+        // Act
+        await FacilityUtilisationDataService.initialiseFacilityUtilisationData(facilityIds, bankId, reportPeriod, requestSource, mockEntityManager);
+
+        // Assert
+        expect(mockSave).toHaveBeenCalledTimes(1);
+        const expectedEntities = [
+          FacilityUtilisationDataEntity.create({
+            id: firstFacilityId,
+            reportPeriod: previousReportPeriod,
+            requestSource,
+            utilisation: firstUtilisation,
+            fixedFee: firstFixedFee,
+          }),
+          FacilityUtilisationDataEntity.create({
+            id: secondFacilityId,
+            reportPeriod: previousReportPeriod,
+            requestSource,
+            utilisation: secondUtilisation,
+            fixedFee: secondFixedFee,
+          }),
+        ];
+        expect(mockSave).toHaveBeenCalledWith(FacilityUtilisationDataEntity, expectedEntities, { chunk: 100 });
+      });
+    });
+  });
+
+  describe('filterOutFacilitiesWithExistingUtilisationData', () => {
+    describe('when all facility ids are associated with existing FacilityUtilisationDataEntities', () => {
```

Comment:
Should we test that `mockFindBy` is called correctly for these test cases?

`toHaveBeenCalledTimes`
`toHaveBeenCalledWith`